astropy/utils/data.py

```diff
@@ -268,6 +268,7 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
                 name_or_obj, cache=cache, show_progress=show_progress,
                 timeout=remote_timeout, sources=sources,
                 http_headers=http_headers)
+        name_or_obj = os.path.expanduser(name_or_obj)
```

Comment:
This is unnecessary if the file is downloaded from URL?
```suggestion
        else:
            name_or_obj = os.path.expanduser(name_or_obj)
```

Reply:
Yes, good point